hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -882,7 +882,7 @@ case class TableMultiWayZipJoin(children: IndexedSeq[TableIR], fieldName: String
         info("TableMultiWayZipJoin: repartitioning children")
         val childRanges = childRVDs.flatMap(_.partitioner.rangeBounds)
         val newPartitioner = RVDPartitioner.generate(childRVDs.head.typ.kType.virtualType, childRanges)
-        childRVDs.map(_.repartition(newPartitioner, ctx))
+        childRVDs.map(_.repartition(newPartitioner, ctx).truncateKey(typ.key.length))
```

Comment:
Ah, I see the problem. But you're computing a new common partitioner with the wrong key, repartitioning each child RVD, then fixing up the partitioner on each child RVD. Better to just compute the right partitioner before doing the repartitioning. I think this should work:
```
val childRanges = childRVDs.flatMap(_.partitioner.coarsenedRangeBounds(typ.key.length))
val newPartitioner = RVDPartitioner.generate(typ.kType, childRanges)
childRVDs.map(_.repartition(newPartitioner, ctx))
```